lib/backend/shadowbackend/client.go

```diff
@@ -82,6 +88,42 @@ func NewClient(config Config, authConfRaw interface{}, stats tally.Scope) (*Clie
 	}, nil
 }
 
+func extractAuthConfigs(config Config, masterAuthConfRaw interface{}) (interface{}, interface{}, error) {
+	aName, sName, err := getBackendNames(config)
+	if err != nil {
+		return nil, nil, err
+	}
+	masterAuth, ok := masterAuthConfRaw.(map[string]interface{})
+	if !ok {
+		return nil, nil, fmt.Errorf("cannot parse master auth config")
+	}
+	aAuth, ok := masterAuth[aName]
+	if !ok {
+		return nil, nil, fmt.Errorf("active backend auth config missing")
+	}
+	sAuth, ok := masterAuth[sName]
+	if !ok {
+		return nil, nil, fmt.Errorf("shadow backend auth config missing")
+	}
+	return aAuth, sAuth, nil
+}
+
+func getBackendNames(config Config) (string, string, error) {
+	if len(config.ActiveClientConfig) != 1 {
+		return "", "", fmt.Errorf("no active backend or more than one active backend configured")
+	}
+	if len(config.ShadowClientConfig) != 1 {
+		return "", "", fmt.Errorf("no shadow backend or more than one shadow backend configured")
+	}
+	var aName string
+	var sName string
+	for aName = range config.ActiveClientConfig { // Map should have only 1 key/value
```

Comment:
Why not just validate the length of the config is 1 and take the value at index 0?

Reply:
ActiveClientConfig (and the shadow one too) are parsed yaml objects, i.e. map[string]map[string]...map[string]interface{}. We can't use indeces, since the object is a map.